pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -172,10 +174,33 @@ func (o *objectWatcherImpl) Update(ctx context.Context, clusterName string, desi
 
 	desireObj, err = o.retainClusterFields(desireObj, clusterObj)
 	if err != nil {
-		klog.Errorf("Failed to retain fields for resource(kind=%s, %s/%s) in cluster %s: %v", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), clusterName, err)
+		klog.Errorf("Failed to retain fields for desired resource(kind=%s, %s/%s) in cluster %s: %v", clusterObj.GetKind(), clusterObj.GetNamespace(), clusterObj.GetName(), clusterName, err)
```

Comment:
The `desired` might not be correct. It's hard to say the retain operation is against the desired or the observed object.

Reply:
I guess since it returns `desireObj`, it shouldn't be a big problem?

Reply:
It's no big deal, but we'd better revert this change, because retain is not for `desired` object.

Reply:
sure